R/zchunk_L2042.resbio_input_irr_mgmt.R

```diff
@@ -36,96 +54,288 @@ module_aglu_L2042.resbio_input_irr_mgmt_DISABLED <- function(command, ...) {
+      L204.GlobalResBio_Mill %>%
+        # rename columns just for binding and working in a single pipeline
+        rename(AgSupplySector = sector.name,
+               AgSupplySubsector = subsector.name,
+               AgProductionTechnology = technology) %>%
+        select(AgSupplySector, AgSupplySubsector, AgProductionTechnology, residue.biomass.production) %>%
+        write_to_all_regions(names = c("region", "AgSupplySector", "AgSupplySubsector", "AgProductionTechnology", "residue.biomass.production"),
+                             GCAM_region_names) %>%
+        filter(! region %in% aglu.NO_AGLU_REGIONS) %>%
+        repeat_add_columns(bind_cols(tibble::tibble(year = MODEL_YEARS))) %>%
+        mutate(colID = "Mill") %>%
+        # bind to processed Forest data
+        bind_rows(For.tmp) %>%
+        # join in base resbio curves
+        repeat_add_columns(select(A_resbio_curves, price, For, Mill)) %>%
+        # assign appropriate fraction harvested based on colID
+        mutate(fract.harvested = if_else(colID == "Mill", Mill, For)) %>%
+        select(-For, -Mill) ->
+        # store in a temporary table for further processing
+        For.Mill.tmp
+    }
+
+    # In base years, replace the "fraction produced" at specified price aglu.PRICE_BIO_FRAC in the model BASE_YEAR,
+    # in order to calibrate resbio production. The fract.harvested for these prices-years are replaced by
+    # the GCAM_region's Base-year fraction of residue biomass produced by region for Forest/Mill as appropriate,
+    # from assumption file A_bio_frac_prod_R
+    For.Mill.tmp %>%
+      filter(price == aglu.PRICE_BIO_FRAC, year %in% BASE_YEARS) %>%
+      left_join_error_no_match(select(A_bio_frac_prod_R, region, For, Mill), by = "region") %>%
+      select(-fract.harvested) %>%
+      # assign appropriate fraction harvested based on colID
+      mutate(fract.harvested = if_else(colID == "Mill", Mill, For)) %>%
+      select(-For, -Mill) %>%
+      bind_rows(filter(For.Mill.tmp, !( price == aglu.PRICE_BIO_FRAC & year %in% BASE_YEARS))) ->
+      For.Mill.tmp2
+
+    # Forest
+    For.Mill.tmp2 %>%
+      filter(colID == "For") %>%
+      select(-colID) ->
+      L204.AgResBioCurve_For
+
+    # Mill
+    For.Mill.tmp2 %>%
+      filter(colID == "Mill") %>%
+      select(-colID) %>%
+      # return correct column names
+      rename(supplysector = AgSupplySector,
+             subsector = AgSupplySubsector,
+             stub.technology = AgProductionTechnology) ->
+      L204.StubResBioCurve_Mill
+
+
+    # AGRICULTURE RESIDUE BIO
+    # 4. Form a table of Agricultural Residue Biomass Paramters by region-glu-year
+    L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      distinct %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      # have to use left_join + na.omit to match  m e r g e  behavior in old DS
```

Comment:
I don't think so; I think the behavior isn't incorrect, we just can't use the most restrictive left_join